nbconvert/exporters/html.py

```diff
@@ -256,8 +260,17 @@ def from_notebook_node(  # type:ignore[explicit-override, override]
         highlight_code = self.filters.get(
             "highlight_code", Highlight2HTML(pygments_lexer=lexer, parent=self)
         )
+        markdown_collection = ""
+        for cell in nb.cells:
+            if cell.cell_type == "markdown":
+                markdown_collection = markdown_collection + cell.source + "\n"
 
         resources = self._init_resources(resources)
+        if resources is None:
+            resources = {}
+        resources.update(
+            {"tableofcontents": extract_titles_from_markdown_input(markdown_collection)}
+        )
```

Comment:
I feel like it would better having this in `_init_resources` as this is the place where we initialize the resources

Reply:
Thanks for the comment. This has been moved to `_init_resources.`

Reply:
Instead of using `update`, you could follow what the rest of the code is doing: `resources["tableofcontents"] = extract_titles_from_markdown_input(markdown_collection)`

Reply:
This was initially was used but without any check that `resources` was not `None`.
Introducing the suggested change leads to this error message
```
nbconvert/exporters/html.py:269: error: Unsupported target for indexed assignment ("Optional[Dict[str, Any]]")  [index]
```

Reply:
I have displaced the `None check` condition back in its initial place, in the `HTMLExporter`